softlayer/vm/softlayer_vm_test.go

```diff
@@ -223,29 +223,37 @@ var _ = Describe("SoftLayerVM", func() {
 
 		Context("found tags in metadata", func() {
 			BeforeEach(func() {
+				softLayerClient.DoRawHttpRequestResponse = []byte("true")
+			})
+
+			It("the tags value is empty", func() {
 				metadataBytes := []byte(`{
 				  "director": "fake-director-uuid",
-				  "name": "fake-director",
-				  "tags": "test, tag, director"
+				  "fake1": "fake-deployment",
+				  "fake2": "buildpack_python"
 				}`)
-
 				metadata = bslvm.VMMetadata{}
 				err := json.Unmarshal(metadataBytes, &metadata)
 				Expect(err).ToNot(HaveOccurred())
 
-				softLayerClient.DoRawHttpRequestResponse = []byte("true")
-			})
-
-			It("the tags value is empty", func() {
-				metadata["tags"] = ""
-				err := vm.SetMetadata(metadata)
+				err = vm.SetMetadata(metadata)
 
 				Expect(err).ToNot(HaveOccurred())
 				Expect(softLayerClient.DoRawHttpRequestResponseCount).To(Equal(0))
 			})
 
 			It("at least one tag found", func() {
-				err := vm.SetMetadata(metadata)
+				metadataBytes := []byte(`{
```

Comment:
We already cover this part of test:
https://github.com/mattcui/bosh-softlayer-cpi/blob/os_reload/softlayer/vm/softlayer_vm_test.go#L204

```go
		Context("no tags found in metadata", func() {
			BeforeEach(func() {
				metadataBytes := []byte(`{
				  "director": "fake-director-uuid",
				  "name": "fake-director"
				}`)

				metadata = bslvm.VMMetadata{}
				err := json.Unmarshal(metadataBytes, &metadata)
				Expect(err).ToNot(HaveOccurred())
			})

			It("does not set any tag values on the VM", func() {
				err := vm.SetMetadata(metadata)

				Expect(err).ToNot(HaveOccurred())
				Expect(softLayerClient.DoRawHttpRequestResponseCount).To(Equal(0))
			})
		})
```